griptape/drivers/vector/base_vector_driver.py

```diff
@@ -36,6 +36,15 @@ class Entry:
         kw_only=True
     )
 
+    def insert_text_artifacts(self, artifacts: list[tuple[TextArtifact, str, Optional[dict]]], **kwargs) -> None:
+        with self.futures_executor as executor:
+            utils.execute_futures_dict({
+                artifact[1]:
+                    executor.submit(self.upsert_text_artifact, artifact[0], artifact[1],
+                                    artifact[2] if len(artifact) > 2 else None, **kwargs)
```

Comment:
This should be similar to the `upsert_text_artifacts` logic except for when the artifact is already in the namespace. Only in that case it should be inserted. Perhaps, we need another abstract method (similar to `load_entry`) that checks whether a vector with a specific ID and namespace already exist in the vector store.